examples/dmm/polyphonic_data_loader.py

```diff
@@ -14,19 +14,46 @@
 """
 
 import os
+import urllib
+from collections import namedtuple
+from urllib.request import urlopen
 
 import numpy as np
 import six.moves.cPickle as pickle
 import torch
 import torch.nn as nn
-from observations import jsb_chorales
 
 from pyro.contrib.examples.util import get_data_directory
 
 
+dset = namedtuple("dset", ["name", "url", "filename", "t_max"])
+
+JSB_CHORALES = dset("jsb_chorales",
+                    "https://d2fefpcigoriu7.cloudfront.net/datasets/polyphonic/jsb_chorales.pickle",
+                    "jsb_chorales.pkl",
+                    160)
+
+PIANO_MIDI = dset("piano_midi",
+                  "https://d2fefpcigoriu7.cloudfront.net/datasets/polyphonic/piano_midi.pickle",
+                  "piano_midi.pkl",
+                  3857)
+
+# FIXME: t_max is too large for pickle.dump
+MUSE_DATA = dset("muse_data",
+                 "https://d2fefpcigoriu7.cloudfront.net/datasets/polyphonic/muse_data.pickle",
+                 "muse_data.pkl",
+                 4273)
```

Comment:
Since `T_max` is much higher for this dataset, the unsparse pickle file gets too large (anything more than 4GB throws an error on Python 3 in mac - [Stackoverflow issue](https://stackoverflow.com/questions/31468117/python-3-can-pickle-handle-byte-objects-larger-than-4gb)). While there are ways to get around this, I think we should think of alternative ways of storing / reading in this data so as to not have to write / load such big objects in the first place. I will need to understand the code a bit better to figure out what to do here. 

@fritzo, @martinjankowiak - Feel free to hack on this branch in the meantime. Note that only this dataset is failing, but the other two are also probably loading in large objects in memory.